### PR TITLE
Fixes for recommended changes from Mozilla #170

### PR DESCRIPTION
**Pull request by Bastlynn**
https://github.com/bs-detector/bs-detector/issues/170

### CODE REVIEW

**Comment by thinkcontext** [comment]
Approved

File: ext/js/background.js [hunk 1]
```diff
@@ -5,14 +5,40 @@
...
 
+
+
+/**
+ * @description Expand a given set of links.
+ * @method expandLinks
+ * @param {string} request The set of links to expand.
+ * @return {promise} A Deferred promise to allow sending after all links processed.
+ */
+ function expandLinks(request, success) {
```
Comment (tomlutzenberger) [inline ext/js/background.js]: Remove 1 space indention

File: ext/js/background.js [hunk 2]
```diff
@@ -79,7 +94,7 @@ xhReq(chrome.extension.getURL('/data/data.json'), function (file) {
             if (e.frameId === 0) {
                 chrome.pageAction.show(e.tabId);
                 domain = url2Domain(e.url);
-                if(domain && siteList[domain]){
+                if (domain && siteList[domain]) {
                   chrome.tabs.sendMessage(e.tabId, {
```
Comment (tomlutzenberger) [inline ext/js/background.js]: Add indention in lines 98-101

File: ext/js/background.js [hunk 3]
```diff
@@ -89,71 +104,51 @@ xhReq(chrome.extension.getURL('/data/data.json'), function (file) {
...
 
 
-// toggle display of the warning UI when the pageAction is clicked
-chrome.pageAction.onClicked.addListener(function (tab) {
+/**
+ * @description Toggle display of the warning UI when the pageAction is clicked.
+ * @method chrome.pageAction.onClicked.addListener
+ * @param {function}
+ */
+ chrome.pageAction.onClicked.addListener(function (tab) {
```
Comment (tomlutzenberger) [inline ext/js/background.js]: Remove 1 space indention

File: ext/js/bs-detector.js [hunk 4]
```diff
@@ -202,13 +204,15 @@ BSDetector.prototype = {
...
         'use strict';
 
-        $.each(this.shorts, function () {
-            var
-                shortLink = 'a[href*="' + $(this) + '"]';
-
-            $(shortLink).each(function () {
-                bsd.toExpand.push($(this).attr('href'));
+        var shorts = this.shorts;
+        var selectors = $('a[href]').filter(function (index, a) {
```
Comment (tomlutzenberger) [inline ext/js/bs-detector.js]: Maybe consider combined `var` statements

File: ext/js/bs-detector.js [hunk 5]
```diff
@@ -627,33 +631,36 @@ BSDetector.prototype = {
  * @param {object} options
  * @param {function} responseCallback
  */
-if(window === window.top || url2Domain(window.location.hostname) == 'twitter.com'){
-  var bsd = new BSDetector();
+if (window === window.top || url2Domain(window.location.hostname) == 'twitter.com') {
```
Comment (tomlutzenberger) [inline ext/js/bs-detector.js]: Why not `===` instead of `==` ?

File: ext/js/utils.js [hunk 6]
```diff
@@ -4,18 +4,50 @@
...
+ * @description Cleanup a url to get the domain out of it.
+ * @method url2Domain
+ * @param {string} url The URL to cleanup.
+ */
+function url2Domain(url) {
+
+    'use strict';
 
-function url2Domain(url){
+   if (url) {
```
Comment (tomlutzenberger) [inline ext/js/utils.js]: 4 space indention

File: ext/js/utils.js [hunk 7]
```diff
@@ -4,18 +4,50 @@
...
+        // Once done loading.
+        if (xhr.readyState === 4) {
+            // Call the right feedback based on response.
+            if (xhr.status === 200 && success) {
+                success(xhr.responseText);
+            } else if (failure) {
+                failure(xhr.responseText);
+            }
+        }
+    }
```
Comment (tomlutzenberger) [inline ext/js/utils.js]: Missing semicolon

**Comment by Bastlynn** [comment]
Done! :)

**Pull request merged**